appender.go

```diff
@@ -866,3 +827,15 @@ func timeFunc(f func()) time.Duration {
 	}
 	return time.Since(t0)
 }
+
+const alphanum = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+
+// NOTE(marclop) until requiring 1.24.0 is possible, then we can switch to
+// rand.Text() from the `crypto/rand` package.
+func randStringBytes(length int) string {
+	buf := make([]byte, length)
+	for i := range buf {
+		buf[i] = alphanum[rand.Intn(len(alphanum))]
+	}
+	return string(buf)
+}
```

Comment:
Instead of generating a random ID, could we use the address of the Appender? i.e. `fmt.Sprintf("%p", a)`